scripts/scraper.ts

```diff
@@ -0,0 +1,341 @@
+        maxBuffer: 100 * 1024 * 1024,
+      });
+
+      const issues = JSON.parse(response) as Issue[];
+
+      if (issues.length === 0) {
+        console.log(`No more ${state} issues found for ${owner}/${repo}`);
+        break;
+      }
+
+      const filteredIssues = issues.filter((issue) => !issue.pull_request);
+      console.log(`Found ${filteredIssues.length} ${state} issues on page ${page}`);
+
+      allIssues = allIssues.concat(filteredIssues);
+      page++;
+
+      await sleep(1000);
+    } catch (error) {
+      console.error(`Error fetching ${state} issues page ${page} for ${owner}/${repo}:`, error);
+      if (error instanceof Error && error.message.includes("rate limit")) {
+        console.log("Rate limit hit, waiting for 60 seconds...");
+        await sleep(60000);
+        continue;
+      }
+      break;
+    }
+  }
+
+  return allIssues;
+}
+
+async function fetchAllIssues(owner: string, repo: string): Promise<Issue[]> {
+  console.log(`\nFetching all issues for ${owner}/${repo}...`);
+
+  const openIssues = await fetchIssuesForState(owner, repo, "open");
+  console.log(`Found ${openIssues.length} open issues`);
+
+  const closedIssues = await fetchIssuesForState(owner, repo, "closed");
+  console.log(`Found ${closedIssues.length} closed issues`);
+
+  const allIssues = [...openIssues, ...closedIssues];
+  console.log(`Total issues found for ${owner}/${repo}: ${allIssues.length}\n`);
+
+  return allIssues;
+}
+
+function calculateTimeToComplete(createdAt: string, closedAt: string | null): string {
+  if (!closedAt) return "Not completed";
+
+  const diffMs = new Date(closedAt).getTime() - new Date(createdAt).getTime();
+
+  const units = [
+    { value: 60, unit: "minute" },
+    { value: 24, unit: "hour" },
+    { value: 7, unit: "day" },
+    { value: 4, unit: "week" },
+    { value: Infinity, unit: "month" },
+  ];
+
+  let time = Math.round(diffMs / (1000 * 60));
+  let unitIndex = 0;
+
+  while (unitIndex < units.length - 1 && time >= units[unitIndex].value) {
+    time = Math.round(time / units[unitIndex].value);
+    unitIndex++;
+  }
+
+  const unit = units[unitIndex].unit;
+  return `${time} ${time === 1 ? unit : unit + "s"}`;
+}
+
+function shuffleArray<T>(array: T[]): T[] {
+  const newArray = [...array];
+  for (let i = newArray.length - 1; i > 0; i--) {
+    // Generate cryptographically secure random number
+    const randomBytes = crypto.randomBytes(4);
+    const j = Math.floor((randomBytes.readUInt32BE(0) / 0x100000000) * (i + 1));
+    [newArray[i], newArray[j]] = [newArray[j], newArray[i]];
+  }
+  return newArray;
+}
+
+function formatContent(issue: Issue, metadata: string): string {
+  return [
+    "### Issue Description",
+    `[${new Date(issue.created_at).toISOString()}] [${issue.user.login}]:\n${issue.body || ""}`,
+    "",
+    "### Issue Details",
+    metadata,
+    "",
+    "### Valid Time Labels",
+    Array.from(VALID_TIME_LABELS).join("\n"),
+  ].join("\n");
+}
```

Comment:
Not sure of the full context of this but you should also include the title as well. Its useful context.